### ISSUE TITLE
where to download the linux version of usb-uart driver program?

### ISSUE THREAD

**Issue by clx7913** [ucb-bar/fpga-zynq#77]
My board is zedboard which contains Cypress CY7C64225 USB-to-UART Bridge device.But I came across a problem that I couldn't find the Linux version of usb-uart driver program.

It seemed that the official wedsite of Cypress only supports the driver in Windows OS,not on the Linux OS.

Need I install another vivado2016.2 on the Windows OS,and complete the rest of work on this OS?

**Comment by clx7913** [comment]
Now I'v installed another vivado2016.2 on the Win7 OS.During the installation of vivado,if you click the related option,it would install the JTAG driver.

As for the Cypress USB-UART driver,we can download the driver program on the official website.

Now I could do the rest of development in the Win7 OS.But I still want to know whether the driver for Linux exists.After all,using two OS is a trouble thing.

**Comment by seldridge** [comment]
More often than not, the Linux driver module is already loaded. On Ubuntu 14.04 and 16.04, I had to do nothing to get the machine to recognize the UART devices. 

You can check what's going on with `dmesg`. Here, I'm plugging two different zedboards and looking at the output of `dmesg`:
```
> dmesg
[2225437.507948] usb 3-5: new full-speed USB device number 7 using xhci_hcd
[2225437.640245] usb 3-5: New USB device found, idVendor=04b4, idProduct=0008
[2225437.640247] usb 3-5: New USB device strings: Mfr=1, Product=2, SerialNumber=4
[2225437.640248] usb 3-5: Product: Cypress-USB2UART-Ver1.0G
[2225437.640250] usb 3-5: Manufacturer: 2012 Cypress Semiconductor
[2225437.640251] usb 3-5: SerialNumber: 046404C94D1E
[2225437.641368] cdc_acm 3-5:1.0: ttyACM0: USB ACM device
[2225598.432942] usb 3-14: new full-speed USB device number 8 using xhci_hcd
[2225598.565245] usb 3-14: New USB device found, idVendor=04b4, idProduct=0008
[2225598.565247] usb 3-14: New USB device strings: Mfr=1, Product=2, SerialNumber=4
[2225598.565248] usb 3-14: Product: Cypress-USB2UART-Ver1.0G
[2225598.565250] usb 3-14: Manufacturer: 2012 Cypress Semiconductor
[2225598.565251] usb 3-14: SerialNumber: 046405030E1E
[2225598.566422] cdc_acm 3-14:1.0: ttyACM1: USB ACM device
```

Similarly, the devices show up with `lsusb`:
```
> lsusb
Bus 003 Device 007: ID 04b4:0008 Cypress Semiconductor Corp. 
Bus 003 Device 008: ID 04b4:0008 Cypress Semiconductor Corp. 
```

The critical part here indicating that some driver picked this up is that you get the lines in the `dmesg` output stating that these were assigned to some tty, specifically, you should see some devices in `/dev` corresponding to your Zedboard UART:
```
> ls -la /dev/ttyACM*
crw-rw---- 1 root dialout 166, 0 Apr  3 12:45 /dev/ttyACM0
crw-rw---- 1 root dialout 166, 1 Apr  3 12:47 /dev/ttyACM1
```

Note: this may not be `/dev/ttyACM*`, it may be something like what is shown in the README.md of `tty.usbserial`, etc. The output of `dmesg` should help you know what to connect to.

The only setup that I had to access these using `screen` was to add myself to the `dialout` group (and log in again). This is expected from the output of `/ls -la /dev/ttyACM*` above where you can see that both of these have no user permissions, but do have `rw` permissions for the `dialout` group.

**Comment by clx7913** [comment]
I tried the "dmesg" and the log was below:
[ 5604.281488] usb 2-1.1: new full-speed USB device number 51 using ehci-pci
[ 5604.379501] usb 2-1.1: New USB device found, idVendor=04b4, idProduct=0008
[ 5604.379506] usb 2-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=4
[ 5604.379509] usb 2-1.1: Product: Cypress-USB2UART-Ver1.0G
[ 5604.379511] usb 2-1.1: Manufacturer: 2012 Cypress Semiconductor
[ 5604.379513] usb 2-1.1: SerialNumber: 021E81591A08
[ 5604.380789] cdc_acm 2-1.1:1.0: ttyACM0: USB ACM device

After I typed "screen /dev/ttyACM0 115200,cs8,-parenb,-cstopb",I didn't succeed in start-up.I got:

ot204.07-dirty (Mar 29 2018 -7:16:13
ard:    Xinx Zq
I2   readyDM:  ECC disabled 256 MiB
MMC:   zynq_sdhci: 0
SF: Ded S2556S_64K wipage size 256 Bytes, er64 KiB, total 32 MiB
*** Warning - baing faulnvirIn:    srial
Out:   serial
Err:   seaNe: b000
Hany  toot  0 
zynq-uboot> TE1 E0
zynq-uboATE1 E0 try 'hlp'
Unnownmmand TE1' - try 'heq-uboot> ~x�~x�~


But when I use SecureCRT on Win7 OS,I got:

PetaLinux v2013.10 ( Yocto 1.4 ) zynq ttyPS0
zynq login:

**Comment by seldridge** [comment]
That's a u-boot prompt. Try `boot` to boot Linux on the ARM core.

**Comment by clx7913** [comment]
Yes,you are right.Thank you for your help!I've logged in the ARM core.

Do you know why after typing `./fesvr-zynq pk hello` it stuck without any response?

root@zynq:~# ./fesvr-zynq pk hello

**Comment by clx7913** [comment]
Do you mean I should build the riscv-fesvr manually? 

I haven't built it in the rocket-chip/riscv-tools/riscv-fesvr/build.

**Comment by clx7913** [comment]
After I saw **RISC-V Toolchain installation completed!** during the riscv-tools' installation a few days ago,I tested it by "spike pk hello" and it displayed "hello",meanning the toolchain is fine.

But just now,I found that "spike pk hello" doesn't work,so I can't say the pk is OK.

**Comment by clx7913** [comment]
I think this might be caused by my uncareful operation: `find . -name target -exec rm -r "{}" ;"` in the root of "fpga-zynq" repo.

The operation may result in some innocent targets contents being deleted.

I'm trying to fix it.I'll reply after accomplishment.

**Comment by clx7913** [comment]
The reason why "spike pk hello" in the rocket-chip directory didn't work was I deleted some innocent targetes cotents in "riscv-tools".I solved it by re-cloned the "riscv-tools" directory.

But after I built the essential files into the SD card and tried "./fesvr-zynq pk hello ",unluckily,it still couldn't throw out any output.

zynq login: root
Password: 
login[818]: root login  on `ttyPS0'

root@zynq:~# ./fesvr-zynq pk hello 

It stuck as before.

**Comment by clx7913** [comment]
@seldridge Hey,now "./fesvr-zynq pk hello" works and could throw out "Hello world!".

I opened the uramdisk.image.gz and replaced the /home/root/hello by another hello program which built by riscv64-unknown-elf-gcc provided by my riscv-toolchain.And then I recompressed the uramdisk.image.gz and loaded the essential files to the SD card.

I think I know why it stuck when I typed "./fesvr-zynq pk hello" before.Maybe in the original uramdisk.image.gz,hello is imcompatible to the fesvr-zynq and/or pk.
